run-tests.php

```diff
@@ -1057,11 +1057,14 @@ function get_binary(string $php, string $sapi, string $sapi_path): ?string
     if (IS_WINDOWS && file_exists("$dir/$sapi.exe")) {
         return realpath("$dir/$sapi.exe");
     }
+    // Sources tree
     if (file_exists("$dir/../../$sapi_path")) {
         return realpath("$dir/../../$sapi_path");
     }
-    if (file_exists("$dir/$sapi")) {
-        return realpath("$dir/$sapi");
+    // Installation tree
+    $inst = str_replace('php', $sapi, basename($php));
```

Comment:
Add a comment that this exists to preserve prefix/suffix of `$php`? It's not super obvious.